### PR TITLE
Add example for pointwise GFRE

### PR DESCRIPTION
**Pull request by agoscinski**
This adds an example notebook using the pointwise GFRE (`pointwise_global_reconstruction_error`)

### CODE REVIEW

File: examples/PlotPointwiseGFRE.ipynb [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+   "outputs": [],
+   "source": [
+    "def compute_standardized_rbf_rkhs_features(features, gamma):\n",
+    "    # standardize features\n",
+    "    features = StandardFlexibleScaler().fit_transform(features)\n",
+    "    \n",
+    "    # compute \\|x - x\\|^2\n",
+    "    squared_distance = ( np.sum(features ** 2, axis=1)[:, np.newaxis]\n",
+    "                                  + np.sum(features ** 2, axis=1)[np.newaxis, :] - 2 * features.dot(features.T) )\n",
+    "    \n",
```
Comment (Luthaf) [inline examples/PlotPointwiseGFRE.ipynb:80]: For readability, I would split these lines a bit more, something like

```py
squared_distance = (
    np.sum(features ** 2, axis=1)[:, np.newaxis]
    + np.sum(features ** 2, axis=1)[np.newaxis, :] 
    - 2 * features.dot(features.T)
)
```

File: examples/PlotPointwiseGFRE.ipynb [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    "    D, A = np.linalg.eigh(kernel)\n",
+    "    \n",
+    "    # retain features associated with an eigenvalue above 1e-9 for denoising\n",
+    "    select_idx = np.where(D > 1e-9)[0]\n",
+    "    \n",
+    "    # compute rkhs features\n",
+    "    rbf_rkhs_features = A[:, select_idx] @ np.diag(np.sqrt(D[select_idx]))\n",
+    "    \n",
+    "    # standardize rkhs features,\n",
+    "    # this step could be ommitted since it is done my the reconstruction measure by default\n",
```
Comment (Luthaf) [inline examples/PlotPointwiseGFRE.ipynb:97]:
```suggestion
    "    # this step could be omitted since it is done buy the reconstruction measure by default\n",
```

File: examples/PlotPointwiseGFRE.ipynb [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    "idx = np.arange(len(power_spectrum_features))\n",
+    "np.random.seed(42)\n",
+    "np.random.shuffle(idx)\n",
+    "train_idx = idx[0::2]\n",
+    "test_idx = idx[1::2]\n",
+    "\n",
+    "print(\"Computing pointwise GFRE...\")\n",
+    "# pointwise global reconstruction error of bispectrum features using power spectrum features\n",
+    "power_spectrum_to_bispectrum_pointwise_gfre = pointwise_global_reconstruction_error(\n",
+    "            power_spectrum_features, bispectrum_features, train_idx=train_idx, test_idx=test_idx)\n",
```
Comment (Luthaf) [inline examples/PlotPointwiseGFRE.ipynb:122]: Same, I would split this a bit for readability. Something like

```py
power_spectrum_to_bispectrum_pointwise_gfre = pointwise_global_reconstruction_error(
    power_spectrum_features, 
    bispectrum_features, 
    train_idx=train_idx,
    test_idx=test_idx
)
```

And similar below

File: examples/PlotPointwiseGFRE.ipynb [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+   "source": [
+    "print(\"GFRE(3-body, 4-body) =\", power_spectrum_to_bispectrum_gfre)\n",
+    "print(\"GFRE(3-body RBF RKHS, 4-body) = \", power_spectrum_rbf_to_bispectrum_gfre)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It can be seen that RBF RKHS features improve the linear reconstruction of the 4-body features (~0.22 in contrast to ~0.19) while also spreading the error for individual samples across a wider span of [0, 0.5] in contrast to [0.17, 0.32]. This indicates that the reconstruction using the RBF RKHS is more particular to the dataset, and is therefore less general."
```
Comment (Luthaf) [inline examples/PlotPointwiseGFRE.ipynb:174]:
```suggestion
    "It can be seen that RBF RKHS features improve the linear reconstruction of the 4-body features (~0.22 in contrast to ~0.19) while also spreading the error for individual samples across a wider span of [0, 0.5] in contrast to [0.17, 0.32]. This indicates that the reconstruction using the RBF RKHS is less generally applicable but instead specific to this dataset."
```

**Review by Luthaf — COMMENTED** [review]
Looks good overall, I have a few improvement suggestions!

File: examples/PlotPointwiseGFRE.ipynb (hunk 4 above)
Comment (agoscinski) [reply examples/PlotPointwiseGFRE.ipynb]: changed

File: examples/PlotPointwiseGFRE.ipynb (hunk 2 above)
Comment (agoscinski) [reply examples/PlotPointwiseGFRE.ipynb]: changed

File: examples/PlotPointwiseGFRE.ipynb (hunk 1 above)
Comment (agoscinski) [reply examples/PlotPointwiseGFRE.ipynb]: changed

**Comment by agoscinski** [comment]
also change usage of random.seed to usage of scikit's random state, since it seems to be against the scikit-learn guidelines https://scikit-learn.org/stable/developers/utilities.html#developers-utils

**Review by Luthaf — APPROVED** [review]
Sorry for the delay, for some reason I missed all notifications on this PR =/
